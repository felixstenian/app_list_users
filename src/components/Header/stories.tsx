import { Meta, Story } from '@storybook/react'
import Header, { HeaderProps } from '.'

export default {
  title: 'Header',
  component: Header,
  arg: {
    children: {
      type: 'string'
    }
  }
} as Meta

export const Default: Story<HeaderProps> = (args) => <Header {...args} />

Default.args = {
  children: 'Header'
}

Default.parameters = {
  layout: 'fullscreen'
}
