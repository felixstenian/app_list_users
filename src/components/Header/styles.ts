import styled, { css } from 'styled-components'

export const Wrapper = styled.header`
  width: 100%;
  height: 10rem;
  display: flex;
  justify-content: center;
  align-items: center;

  ${({ theme }) => css`
    background-color: ${theme.colors.mainBg};
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.xxlarge};
  `}
`
