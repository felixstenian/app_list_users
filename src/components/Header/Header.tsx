import * as Style from './styles'

export type HeaderProps = {
  children: React.ReactNode
}

const Header = ({ children }: HeaderProps) => (
  <Style.Wrapper>{children}</Style.Wrapper>
)

export default Header
