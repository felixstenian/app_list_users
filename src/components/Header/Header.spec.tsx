import { screen } from '@testing-library/react'

import { renderWithTheme } from 'utils'
import Header from '.'

describe('<Header />', () => {
  it('1. should render the reading', () => {
    const { container } = renderWithTheme(
      <Header>
        <h1>Header</h1>
      </Header>
    )

    expect(screen.getByText(/Header/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('2. should render the header with the given props', () => {
    const { container } = renderWithTheme(
      <Header>
        <h1>Logo</h1>
      </Header>
    )

    expect(screen.getByRole('heading', { name: /Logo/i })).toBeInTheDocument()
    expect(container.firstChild).toHaveStyle({ width: '100%', height: '10rem' })
  })
})
