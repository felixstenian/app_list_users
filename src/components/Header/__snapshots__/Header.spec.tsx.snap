// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> 1. should render the reading 1`] = `
.c0 {
  width: 100%;
  height: 10rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #06092B;
  color: #FAFAFA;
  font-size: 2.8rem;
}

<header
  class="c0"
>
  <h1>
    Header
  </h1>
</header>
`;
